### PR TITLE
feat(atomic,headless): add highlights, atomic-result-excerpt & atomic-result-uri components

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-305
Screen Shot 2021-02-19 at 9 26 48 AM

### CODE REVIEW

File: packages/headless/src/utils/highlight.ts [hunk 1]
```diff
@@ -11,6 +12,13 @@ export interface HighlightKeyword {
   length: number;
 }
 
+export type ResultHighlights =
```
Comment (samisayegh) [inline packages/headless/src/utils/highlight.ts:15]: Hmm, it does not look right to have this type exported from headless, since it is only used inside atomic?

It would make more sense to include such a type if we exported a `highlightResult` utility, e.g.

```
highlightResult<T extends Result>(result: T, highlight: ResultHighlight, delimiters) {}
```
but I think having a dedicated function for each kind of highlight would be even more friendly

```
highlightResultExcerpt<T extends Result>(result: T, delimiters) {}
```

tl;dr I don't think we should have such a type in headless unless I am missing something?

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/src/components/result-template-components/atomic-result-uri/atomic-result-uri.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  ResultContext,
+  ResultContextRenderer,
+} from '../result-template-decorators';
+
+/**
+ * The ResultUri component displays the URI, or path, to access a result.
+ * @part result-uri - The result uri
+ */
+@Component({
+  tag: 'atomic-result-uri',
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/components/result-template-components/atomic-result-uri/atomic-result-uri.tsx:13]: Could use the JSUI name `atomic-result-printable-uri`

File: packages/atomic/src/components/result-template-components/atomic-result-value/atomic-result-value.tsx [hunk 3]
```diff
@@ -1,30 +1,73 @@
...
       this.result,
       this.value
-    );
-    if (resultValue !== null) {
-      return resultValue;
+    )}`;
+  }
+
+  private renderWithHighlights() {
+    try {
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/components/result-template-components/atomic-result-value/atomic-result-value.tsx:37]: highlightString throws an error if the options are invalid
